bundle-workflow/README.md

```diff
@@ -1,14 +1,38 @@
 - [OpenSearch Bundle Workflow](#opensearch-bundle-workflow)
+    - [How it works](#how-it-works)
     - [Build from Source](#build-from-source)
         - [Custom Build Scripts](#custom-build-scripts)
     - [Assemble the Bundle](#assemble-the-bundle)
         - [Custom Install Scripts](#custom-install-scripts)
     - [Signing Artifacts](#signing-artifacts)
+    - [Plugin Onboarding](#plugin-onboarding)
 
 ## OpenSearch Bundle Workflow
 
 This workflow builds a complete OpenSearch bundle from source. You can currently build 1.0, 1.1 and 1.1-SNAPSHOT.
 
+### How it works
```

Comment:
@setiah @CEHENKLE This is very much design doc turned into documentation and does a great job at explaining what we have in the best place possible, next to the code, therefore I don't see why we need another doc checked into the repo that duplicates the code and its documentation